sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -772,7 +768,7 @@ AllocaCommandBase *Scheduler::GraphBuilder::getOrCreateAllocaForReq(
           // new one. There could be situations when we could setup link with
           // "not" current allocation, but it will require memory copy.
           // Can setup link between cl and host allocations only
-          if (Queue->is_host() != Record->MCurContext->is_host()) {
+          if ((Context != nullptr) != (Record->MCurContext != nullptr)) {
```

Comment:
```suggestion
          if ((Context == nullptr) != (Record->MCurContext == nullptr)) {
```
Nit: this is probably subjective, but I find this slightly easier to read.